alonzo/formal-spec/protocol-parameters.tex

```diff
@@ -182,9 +205,8 @@ \subsection{Script Evaluation Cost Model and Prices}
   %
   \begin{align*}
     & \fun{getLanguageView} \in \PParams \to \Language \to \LangDepView \\
-    & \fun{getLanguageView}~\var{pp}~\PlutusVI = \var{costmdls}~\mapsto~ (\fun{costmdls}~{pp}~\{\PlutusVI\})
+    & \fun{getLanguageView}~\var{pp}~\PlutusVI = (\fun{serialize}~\PlutusVI, ~ (\fun{serializeEncoding}~(\fun{costmdls}~{pp}~\{\PlutusVI\})))
```

Comment:
I really don't think this should have been changed. The only reason why it isn't implemented like this is because Haskell can't properly express it. This adds a bunch of complexity and extra primitive functions that don't add any extra clarity.

Reply:
How do you propose we make the types match up? `LangDepView` should now be a type common to all languages, ie just a bytestring of all the relevant info

Reply:
I think regarding the types it was just fine before. In the Shelley spec, there are also `PParams` and `PParamsUpdate` types that don't work like that in Haskell. There is no reason to limit ourselves to what's possible to express in Haskell.

Adding the `PlutusVI` tag is probably necessary, but again, no byte strings necessary.

Reply:
As a compromise, I propose we make the type `LangDepView` abstract, and say that we require it to have a `hash` function and that it should be able to hold the data we put into it in `getLanguageView`.

Reply:
i forget, is this still your opinion or are you ok with using `serialize` (which I already used in defining `serSize` in ShelleyMA)

Reply:
I think `serialize` is fine. I'd prefer it if it was just a single `ByteString` though that contains everything, and no `serializeEncoding` function.